dask/dataframe/io.py

```diff
@@ -447,27 +447,42 @@ def to_hdf(df, path_or_buf, key, mode='a', append=False, complevel=0,
 
     # when writing to the same file in multiple tasks we should lock
     if lock is None:
-        if not single_node:
+        if not single_node or not single_file:
             lock = True
         else:
             lock = False
 
-    # if get not specified default to sync if we're writing to a single entity
-    # if we're writing to multiple files, default to multiprocessing
-    # by default hdf5 is not thread safe, so threaded is not an option
-    if get is None:
-        if not single_file:
-            get = dask.multiprocessing.get
-        else:
-            get = get_sync
+    if lock is True:
+        lock = Lock()
+
+    def _pd_to_hdf(*args, **kwargs):
+        lock = kwargs.pop('lock')
+
+        if lock:
+            go = False
+            while not go:
+                print(lock)
+                go = lock.acquire(blocking=False)
+                print(go)
+            #lock.acquire()
+            print("acquired")
+        try:
+            pd_to_hdf(*args, **kwargs)
+        finally:
+            if lock:
+                lock.release()
+                print("released")
+
+        return None
```

Comment:
Dask.dataframe depends on partd which depends on locket.  You might want to
try locket.lock_file

On Mon, Jun 27, 2016 at 2:39 PM, Nir <notifications@github.com> wrote:

> In dask/dataframe/io.py
> <https://github.com/dask/dask/pull/1293#discussion_r68661234>:
>
> > +        if lock:
> > +            go = False
> > +            while not go:
> > +                print(lock)
> > +                go = lock.acquire(blocking=False)
> > +                print(go)
> > +            #lock.acquire()
> > +            print("acquired")
> > +        try:
> > +            pd_to_hdf(*args, **kwargs)
> > +        finally:
> > +            if lock:
> > +                lock.release()
> > +                print("released")
> > +
> > +        return None
>
> Sorry, the prints an go were obviously leftovers (from trying to use
> multiprocessing.Lock, actually).
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/dask/dask/pull/1293/files/4e7f0722af962655360b5c7b8b6b1b869737b5fa..62c7ecd0f731fd39bec25e563e244daf69158bfd#r68661234>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AASszLF-uND7l7CCQ_smChNi-fnTBm7Pks5qQEL1gaJpZM4I2sWs>
> .
>